### ISSUE TITLE
RAND_DRBG errors while using secure heap

### ISSUE THREAD

**Issue by buckythebear** [openssl/openssl#9495]
Hi,

I have a problem using secure heap with RAND_DRBG in OpenSSL 1.1.1.
If I pass secure heap size less than 65536 bytes to CRYPTO_secure_malloc_init() - I can't generate keys, get random etc.

Here is an example:
```
#include <openssl/crypto.h>
#include <openssl/ec.h>
#include <openssl/err.h>
#include <openssl/evp.h>

int main() {
	int r = CRYPTO_secure_malloc_init(32768, 32);
	if (!r) {
		printf("CRYPTO_secure_malloc_init failed\n");
		return -1;
	}

	EC_KEY *key = EC_KEY_new();
	if (!key) {
		CRYPTO_secure_malloc_done();
		printf("EC_KEY_new failed\n");
		return -1;
	}

	EC_GROUP *ecgroup = EC_GROUP_new_by_curve_name(NID_secp256k1);
	if (!ecgroup) {
		EC_KEY_free(key);
		CRYPTO_secure_malloc_done();
		printf("EC_GROUP_new_by_curve_name failed\n");
		return -1;
	}

	r = EC_KEY_set_group(key, ecgroup);
	if (!r) {
		EC_GROUP_free(ecgroup);
		EC_KEY_free(key);
		CRYPTO_secure_malloc_done();
		printf("EC_KEY_set_group failed\n");
		return -1;
	}

	r = EC_KEY_generate_key(key);
	if (!r) {
		printf("EC_KEY_generate_key failed\n");

		unsigned long err = ERR_get_error();
		while (err != 0) {
			printf("%s\n", ERR_error_string(err, NULL));
			err = ERR_get_error();
		}

		EC_GROUP_free(ecgroup);
		EC_KEY_free(key);
		CRYPTO_secure_malloc_done();
		return -1;
	}

	EC_GROUP_free(ecgroup);
	EC_KEY_free(key);
	CRYPTO_secure_malloc_done();
	return 0;
}
```

EC_KEY_generate_key fails due to malloc errors in DRBG. If I change secure heap size to 65536 bytes - everything works fine.
AFAIK there is a 64 kilobytes limit on mlock-ed memory per unprivileged process, so if my application is using openssl with secure heap I can not mlock any memory in it.
What is an expected amount of secure memory required by RAND_DRBG? Can't there be any memory leak, causing this much memory required?

**Comment by mspncp** [comment]
Your issue sounds like the one fixed in #9424 recently, see e8d866dcb170dbe22dcda418cd5df655f67dbc0b.

Could you please try whether your issue still persists with a current build from the 
[OpenSSL_1_1_1-stable](https://github.com/openssl/openssl/commits/OpenSSL_1_1_1-stable) branch?

**Comment by buckythebear** [comment]
Everything works fine on OpenSSL_1_1_1-stable branch.
